Mathlib/Geometry/Euclidean/Sphere/Tangent.lean

```diff
@@ -178,6 +188,49 @@ lemma IsTangentAt.isTangent {s : Sphere P} {p : P} {as : AffineSubspace ℝ P}
   · rw [center_mem_orthRadius_iff] at hsp
     rwa [← hsp] at hs
 
+lemma IsTangent.infDist_eq_radius {s : Sphere P} {as : AffineSubspace ℝ P} (h : s.IsTangent as) :
+    Metric.infDist s.center as = s.radius := by
+  obtain ⟨p, h⟩ := h
+  refine le_antisymm ?_ ?_
+  · convert Metric.infDist_le_dist_of_mem h.mem_space
+    rw [mem_sphere'.1 h.mem_sphere]
+  · rw [Metric.infDist_eq_iInf]
+    have : Nonempty as := ⟨⟨p, h.mem_space⟩⟩
+    refine le_ciInf fun x ↦ le_of_sq_le_sq ?_ dist_nonneg
+    rw [dist_comm, h.dist_sq_eq_of_mem x.property, le_add_iff_nonneg_right]
+    exact sq_nonneg _
+
+lemma dist_orthogonalProjection_eq_radius_iff_isTangentAt {s : Sphere P} {as : AffineSubspace ℝ P}
+    [Nonempty as] [as.direction.HasOrthogonalProjection] :
+    dist s.center (orthogonalProjection as s.center) = s.radius ↔
```

Comment:
Is this
```suggestion
    Metric.infDist s.center as = s.radius ↔
```
which is then more consistent with `infDist_eq_radius`

Reply:
The current form of this lemma is deliberate, since it's talking about the projection as the point of tangency.

Reply:
I think what I'm really after here is `infDist_eq_radius` as an `Iff`, which I believe is true but perhaps easier to prove separately.

Reply:
I've added that as deduced from other lemmas already present. Note that it requires the orthogonal projection to exist, which `infDist_eq_radius` doesn't.

Reply:
Is that an artifact of how you proved it (in which case a TODO would be fine), or a necessary assumption?

Reply:
It's a necessary assumption (when the subspace doesn't have orthogonal projections, the infimum may not be attained, and if it's not attained, there is no point of tangency).